librz/core/cmd_egg.c

```diff
@@ -131,166 +90,182 @@ static int cmd_egg_compile(RzEgg *egg) {
 	return ret;
 }
 
-RZ_IPI int rz_cmd_egg(void *data, const char *input) {
-	RzCore *core = (RzCore *)data;
+static RzEgg *rz_core_egg_setup(RzCore *core) {
 	RzEgg *egg = core->egg;
-	char *oa, *p;
-	rz_egg_setup(egg,
-		rz_config_get(core->config, "asm.arch"),
-		core->rasm->bits, 0,
-		rz_config_get(core->config, "asm.os")); // XXX
-	switch (*input) {
-	case 's': // "gs"
-		// TODO: pass args to rz_core_syscall without vararg
-		if (input[1] == ' ') {
-			RzBuffer *buf = NULL;
-			const char *ooaa = input + 2;
-			while (IS_WHITESPACE(*ooaa) && *ooaa)
-				ooaa++;
-			oa = strdup(ooaa);
-			p = strchr(oa + 1, ' ');
-			if (p) {
-				*p = 0;
-				buf = rz_core_syscall(core, oa, p + 1);
-			} else {
-				buf = rz_core_syscall(core, oa, "");
-			}
-			free(oa);
-			if (buf) {
-				showBuffer(buf);
-			}
-			egg->lang.nsyscalls = 0;
-		} else {
-			eprintf("Usage: gs [syscallname] [parameters]\n");
-		}
-		break;
-	case ' ': // "g "
-		if (input[1] && input[2]) {
-			rz_egg_load(egg, input + 2, 0);
-			if (!cmd_egg_compile(egg)) {
-				eprintf("Cannot compile '%s'\n", input + 2);
-			}
-		} else {
-			eprintf("wat\n");
+	const char *arch = rz_config_get(core->config, "asm.arch");
+	const char *os = rz_config_get(core->config, "asm.os");
+	int bits = rz_config_get_i(core->config, "asm.bits");
+
+	if (!rz_egg_setup(egg, arch, bits, 0, os)) {
+		RZ_LOG_ERROR("Cannot setup shellcode compiler for chosen configuration");
+		return NULL;
+	}
+	return egg;
+}
+
+RZ_IPI RzCmdStatus rz_egg_compile_handler(RzCore *core, int argc, const char **argv) {
+	RzEgg *egg = rz_core_egg_setup(core);
+	if (!egg) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	if (argc > 1) {
+		rz_egg_load(egg, argv[1], 0);
+	}
+	if (!rz_core_egg_compile(egg)) {
+		RZ_LOG_ERROR("Cannot compile the shellcode");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_egg_config_handler(RzCore *core, int argc, const char **argv) {
+	RzEgg *egg = rz_core_egg_setup(core);
+	if (!egg) {
+		return RZ_CMD_STATUS_ERROR;
+	}
+	if (argc > 1) {
+		// Set the config option
+		RzList *l = rz_str_split_duplist_n(argv[1], "=", 1, false);
```

Comment:
I think this is wrong. You potentially have an array of arguments, not just `argv[1]`, so each one of them should be split by `=`, not just `argv[1]`.